src/DataStructures/SpinWeighted.hpp

```diff
@@ -36,13 +138,31 @@ SPECTRE_ALWAYS_INLINE
   return {lhs.data + rhs.data};
 }
 template <typename T>
-SPECTRE_ALWAYS_INLINE SpinWeighted<T, 0> operator+(
-    const SpinWeighted<T, 0>& lhs, const T& rhs) noexcept {
+SPECTRE_ALWAYS_INLINE
+    SpinWeighted<decltype(std::declval<T>() + std::declval<T>()), 0>
```

Comment:
Are you doing this to allow expression templates to be wrapped in `SpinWeighted`? If so, could you please add comments since I expect someone looking at this in the future will be confused.